python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py

```diff
@@ -0,0 +1,388 @@
+
+    Examples
+    --------
+    >>> import dgl
+    >>> from dgl.dataloading import NeighborSampler, DataLoader
+    >>> from dgl.nn import CuGraphRelGraphConv
+    >>> import torch.nn.functional as F
+    ...
+    >>> device = 'cuda'
+    >>> fanouts = [5, 6]
+    >>> sampler = NeighborSampler(fanouts)
+    >>> dataloader = DataLoader(g, train_nid, sampler,
+    ...     device=device, batch_size=1024)
+    >>> conv1 = CuGraphRelGraphConv(in_dim, h_dim, num_rels, fanouts[0],
+    ...     regularizer='basis', num_bases=10).to(device)
+    >>> conv2 = CuGraphRelGraphConv(h_dim, out_dim, num_rels, fanouts[1],
+    ...     regularizer='basis', num_bases=10).to(device)
+    >>> for input_nodes, output_nodes, blocks in dataloader:
+    ...     h = conv1(blocks[0], x, blocks[0].edata[dgl.ETYPE])
+    ...     h = F.relu(h)
+    ...     h = conv2(blocks[1], h, blocks[1].edata[dgl.ETYPE])
+    """
+    def __init__(
+        self,
+        in_feat,
+        out_feat,
+        num_rels,
+        fanout,
+        regularizer=None,
+        num_bases=None,
+        bias=True,
+        activation=None,
+        self_loop=True,
+        dropout=0.0,
+        layer_norm=False,
+    ):
+        super().__init__()
+        self.in_feat = in_feat
+        self.out_feat = out_feat
+        self.num_rels = num_rels
+        self.fanout = fanout
+
+        # regularizer (see dgl.nn.pytorch.linear.TypedLinear)
+        if regularizer is None:
+            self.W = nn.Parameter(th.Tensor(num_rels, in_feat, out_feat))
+            self.coeff = None
+        elif regularizer == "basis":
+            if num_bases is None:
+                raise ValueError(
+                    'Missing "num_bases" for basis regularization.'
+                )
+            self.W = nn.Parameter(th.Tensor(num_bases, in_feat, out_feat))
+            self.coeff = nn.Parameter(th.Tensor(num_rels, num_bases))
+            self.num_bases = num_bases
+        else:
+            raise ValueError(
+                f"Supported regularizer options: 'basis', "
+                f"but got {regularizer}"
+            )
+        self.regularizer = regularizer
+        self.reset_parameters()
+
+        # others
+        self.bias = bias
+        self.activation = activation
+        self.self_loop = self_loop
+        self.layer_norm = layer_norm
+
+        # bias
+        if self.bias:
+            self.h_bias = nn.Parameter(th.Tensor(out_feat))
+            nn.init.zeros_(self.h_bias)
+
+        # layer norm
+        if self.layer_norm:
+            self.layer_norm_weight = nn.LayerNorm(
+                out_feat, elementwise_affine=True
+            )
+
+        # weight for self_loop
+        if self.self_loop:
+            self.loop_weight = nn.Parameter(th.Tensor(in_feat, out_feat))
+            nn.init.xavier_uniform_(
+                self.loop_weight, gain=nn.init.calculate_gain("relu")
+            )
+
+        self.dropout = nn.Dropout(dropout)
+
+    def reset_parameters(self):
```

Comment:
Typically `reset_parameters` allows users to reset all learnable parameters while this currently only handles `self.coeff` and `self.W`. 

- We can either do these directly in `__init__` if this is only used by `__init__`
- Or expand it to cover all learnable parameters

Reply:
great suggestion. The initialization calls now go into __init__.